src/modules/iotjs_module_adc.c

```diff
@@ -231,12 +232,11 @@ JHANDLER_FUNCTION(AdcConstructor) {
 
 
 JHANDLER_FUNCTION(Read) {
-  DJHANDLER_CHECK_THIS(object);
+  JHANDLER_DECLARE_THIS_PTR(adc, adc);
   DJHANDLER_CHECK_ARG_IF_EXIST(0, function);
 
   const iotjs_jval_t* jcallback = JHANDLER_GET_ARG_IF_EXIST(0, function);
   const iotjs_jval_t* jadc = JHANDLER_GET_THIS(object);
```

Comment:
You can write `const iotjs_jval_t* jadc = iotjs_jobjectwrap_jobject(adc);` instead, but this is also a good solution. @daeyeon What do you think? Which would be better?